Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -269,80 +382,47 @@ void MediaSource::completeSeek()
     auto seekTarget = *m_pendingSeekTarget;
     m_pendingSeekTarget.reset();
 
-    Ref<MediaTimePromise> promise = SourceBuffer::ComputeSeekPromise::all(WTF::map(*m_activeSourceBuffers, [&](auto&& sourceBuffer) {
+    MediaTimePromise::AutoRejectProducer producer(PlatformMediaError::SourceRemoved);
+    Ref promise = producer.promise();
+
+    scriptExecutionContext()->enqueueTaskWhenSettled(SourceBuffer::ComputeSeekPromise::all(WTF::map(*m_activeSourceBuffers, [&](auto&& sourceBuffer) {
         return sourceBuffer->computeSeekTime(seekTarget);
-    }))->whenSettled(m_dispatcher, [time = seekTarget.time, protectedThis = Ref { *this }] (auto&& results) mutable {
+    })), TaskSource::MediaElement, [producer = WTFMove(producer), weakThis = WeakPtr { *this }, this, time = seekTarget.time](auto&& results) {
+        if (!weakThis || isClosed())
+            return;
```

Comment:
Don't you need to wrap `weakThis` in a `protectedThis` here? Otherwise there's nothing keeping `this` alive during the run of this method.

Reply:
There's no external call that could make the MEdiaSource be deleted; and while the task is running, it can't be deleted. The MediaSource is only ever accessed on the context's dispatcher